<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta name="viewport" content="width=device-width, initial-scale=1">



<title>POSTER_EUROBIOC_20250805.knit</title>







<!--
Font-awesome icons ie github or twitter
-->
<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.8.1/css/all.css" integrity="sha384-50oBUHEmvpQ+1lW4y57PTFmhCaXp0ML5d60M1M7uH2+nqUivzIebhndOJK28anvf" crossorigin="anonymous">
<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.8.1/css/brands.css" integrity="sha384-n9+6/aSqa9lBidZMRCQHTHKJscPq6NW4pCQBiMmHdUCvPN8ZOg2zJJTkC7WIezWv" crossorigin="anonymous">

<!--
Google fonts api stuff
-->
<link href='https://fonts.googleapis.com/css?family=Lato' rel='stylesheet'>
<link href='https://fonts.googleapis.com/css?family=Lato' rel='stylesheet'>

<!--
Here are the required style attributes for css to make this poster work :)
-->
<style>
@page {
size: 37.40in 73.23in;
margin: 0;
padding: 0;
}
body {
margin: 0px;
padding: 0px;
width: 37.40in;
height: 73.23in;
text-align: justify;
font-size: 35pt;
line-height: 1.05;
}
/* RMarkdown Class Styles */
/* center align leaflet map,
from https://stackoverflow.com/questions/52112119/center-leaflet-in-a-rmarkdown-document */
.html-widget {
margin: auto;
position: sticky;
margin-top: 2cm;
margin-bottom: 2cm;
}
.leaflet.html-widget.html-widget-static-bound.leaflet-container.leaflet-touch.leaflet-fade-anim.leaflet-grab.leaflet-touch-drag.leaflet-touch-zoom {
position: sticky;
width: 100%;
}
pre.sourceCode.r {
background-color: #dddddd40;
border-radius: 4mm;
padding: 4mm;
width: 75%;
margin: auto;
margin-top: 1em;
margin-bottom: 1em;
/* align-items: center; */
}
code.sourceCode.r{
background-color: transparent;
font-size: 20pt;
border-radius: 2mm;
}
code {
font-size: 25pt;
font-family: monospace;
background-color: #32677124;
color: #55345e;
padding: 1.2mm;
line-height: 1;
border-radius: 2mm;
}
caption {
margin-bottom: 10px;
font-size: 20pt;
font-style: italic;
}

tbody tr:nth-child(odd) {
    background-color: #32677120;
}
.table>thead>tr>th, .table>tbody>tr>th, .table>tfoot>tr>th, .table>thead>tr>td, .table>tbody>tr>td, .table>tfoot>tr>td{
  border-spacing: 0;
  font-size: 40%;
  border-style: none;
  padding-top: 15px;
  padding-bottom: 15px;
  padding-right: 1em;
  padding-left: 1em;
  line-height: 1em;
}
table {
  margin: auto;
}
th {
  padding-left: 5mm;
  padding-right: 5mm;
}
.caption {
font-size: 20pt;
font-style: italic;
padding-top: 0;
}
.references {
font-size: 32pt;
line-height: 90%;
}
/* Create three unequal columns that floats next to each other */
.column {
float: left;
padding: 0px;
}
.outer {
width: 37.40in;
height: calc(73.23in *  (1 - 0.2 - 4.0e-2 - 0.01) );
-webkit-column-count: 3; /* Chrome, Safari, Opera */
-moz-column-count: 3; /* Firefox */
column-count: 3;
-webkit-column-fill: auto;
-moz-column-fill: auto;
column-fill: auto;
column-gap: 0;
padding-left: 0cm;
padding-right: 0cm;
/* -webkit-column-rule-width: 50%;
-moz-column-rule-width: 50%;
column-rule-width: 50%; */
-webkit-column-rule-style: none;
-moz-column-rule-style: none;
column-rule-style: none;
-webkit-column-rule-color: black;
-moz-column-rule-color: black;
column-rule-color: black;
background-color: #ffffff;
font-family: Lato;
margin-top: calc(73.23in *  0.2 );
padding-top: 1em;
padding-bottom: 1em;
}
span.citation {
  color: #326771;
  font-weight: bold;
}
a {
text-decoration: none;
color: #326771;
}
#title {
font-size: 72pt;
text-align: left;
margin: 0;
line-height: 98%;
border-bottom: 0;
font-weight: normal;
background: 0;
}
#author {
color: #55345e;
margin: 0;
line-height: 85%;
font-size: 52pt;
}
#affiliation {
padding-top: 0.1em;
color: #00000080;
font-style: italic;
font-size: 32pt;
margin: 0;
}
sup {
color: #49817b;
}
.affiliation sup {
font-size: 20px;
}
.author {
text-align: left;
}
.author sup {
font-size: 30px;
}
.author_extra {
color: #326771;
margin: 0;
line-height: 85%;
font-size: 42pt;
text-align: left;
}
.outer h1, h2, h3, h4, h5, h6 {
text-align: center;
margin: 0;
font-weight: bold;
}
.section h1 {
  text-align:center;
  padding-bottom:5px;
  background:
    linear-gradient(
      to left,
      #ffffff 1%,
      #ffffff 20%,
      #55345e75 33%,
      #55345e 50%,
      #55345e75 66%,
      #ffffff 80%,
      #ffffff 99%
    )
    left
    bottom
    #ffffff
    no-repeat;
  background-size:100% 5px ;
  margin-top: 0.5em;
  margin-bottom: 0.5em;
}
.outer h2 {
text-align: center;
}
.outer p, .level2 {
color: #000000;
}
.outer ol {
padding-left: 8%;
padding-right: 8%;
text-align: left;
}
.main {
width: 37.40in;
height: calc(73.23in * 0.2);
position: absolute;
background-color: #55345e;
color: #FFFFFF100;
font-family: Lato;
background-image: linear-gradient(#55345e 50%,#326771);
}
.main strong {
color: #FFFFFF;
}
.main strong > sup {
color: #FFFFFF;
}
.main sup {
color: #FFFFFF100;
}
#main-img-left {
width: 10%;
left: 0.5in;
bottom: 0.2in;
position: absolute;
}
#main-img-center {
width: 10%;
left: calc(37.40in * 0.45);
bottom: 0.5in;
position: absolute;
}
#main-img-right {
width: 10%;
right: 0.5in;
bottom: 0.2in;
position: absolute;
}
.main p {
font-size: 82pt;
font-family: Lato;
text-align: center;
margin: 0;
position: absolute;
top: 50%;
-ms-transform: translateY(-50%);
transform: translateY(-50%);
margin-left: 1em;
}
.fab {
color: #00000030;
font-size: 25px;
}
.twitter, i {
color: #00000030;
font-size: 35px;
text-decoration: none;
}
a.email {
text-decoration: none;
color: #00000030;
font-size: 35px;
}
.envelope {
color: #00000030;
font-size: 5px;
text-decoration: none;
}
.poster_wrap {
width: 37.40in;
height: 73.23in;
padding: 0cm;
}
.main_bottom {
width: 37.40in;
height: calc(73.23in * 4.0e-2);
margin-top: calc(73.23in * (1 - 4.0e-2));
position: absolute;
background-color: #55345e;
background-image: linear-gradient(#326771 10%, #55345e);
}
.section {
  padding-left: 10mm;
  padding-right: 10mm;
}
span > #tab:mytable {
  font-weight: bold;
}
.orcid img {
  width: 3%;
}
.emphasis {
  background-color: #008080;
  color: #ffffff;
  border: solid #0b2045 3mm;
  margin: 1em;
  padding-left: 0;
  padding-right: 0;
}
.emphasis h1 {
  font-weight: bold;
  background: none;
  background-color: #0b2045;
  padding-bottom: 5mm;
  padding-top: 1mm;
  margin-top: -1mm;
  margin-right: -1mm;
  margin-left: -1mm;
}
.emphasis blockquote {
  border: 0;
}
.emphasis ol {
  padding: 0;
  padding-left: 8%;
  font-size: 100%;
  font-weight: bold;
}
.emphasis p {
  color: #ffffff;
}
</style>
</head>
<body>


<div class="poster_wrap">

<div class="column outer">
<div class="section">

<h3 id="author" class="author">

<strong>Georgina Fuentes-Páez</strong><sup> 1, <a class="orcid" href="https://orcid.org/0009-0003-9417-7148"><img src="https://raw.githubusercontent.com/brentthorne/posterdown/master/images/orcid.jpg"></a></sup><br>

<a class='envelope'><i class="fas fa-envelope"></i></a> <a href="mailto:georgina.fuentes@upf.edu" class="email">georgina.fuentes@upf.edu</a> <br>
   </h3>

<h5 id="author_extra", class="author_extra">
 Nacho Molina<sup>2</sup>
 Mireia Ramos-Rodríguez<sup>1</sup>
 Lorenzo Pasquali<sup>1</sup>
</h5>


<p id="affiliation" class="affiliation">
<sup>1</sup> Endocrine Regulatory Genomics, Department of Experimental &amp; Health Sciences, University Pompeu Fabra, 08003, Barcelona, Spain.<br> <sup>2</sup> Institut de génétique et de biologie moléculaire et cellulaire (IGBMC), University of Strasbourg, 67404, Illkirch, France
</p>
</div>

<style type="text/css">
.table caption {
    font-size: 30pt
}
</style>
<style>
  /* Wrapper for all three logos */
  #logo-container {
    display: flex;
    justify-content: space-between; /* space them evenly */
    align-items: center;
  }

  #main-img-left {
    width: 18%;
  }

  #main-img-center {
    width: 37%;
   margin-left: -23%; /* shift left */
  }

  #main-img-right {
    width: 38%;
  }

  /* Headers */
  h1 {
    font-size: 45pt;
  }
  h2,h3,h4,h5,h6 {
    font-size: 36pt;
  }

  /* Change space between figure and captions */
  p.caption {
    margin-top: 0;
    margin-bottom: 0;
  }
</style>
<div id="background" class="section level1">
<h1>Background</h1>
<p>Single-cell RNA and ATAC sequencing allow high-resolution profiling of gene expression and chromatin accessibility, revealing cell-type-specific regulatory programs. Yet, tools to quantify specificity across multiple cell types are still limited <span class="citation">Lim et al. (2024)</span>.</p>
<p><strong>SPICEY</strong> (<strong>SP</strong>ecificity <strong>I</strong>ndex for <strong>C</strong>oding and <strong>E</strong>pigenetic activit<strong>Y</strong>) is the first R package to quantify cell-type specificity from single-cell multi-omic data, offering an intuitive pipeline for analyzing and visualizing transcriptomic and epigenomic specificity.</p>
</div>
<div id="methods" class="section level1">
<h1>Methods</h1>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:workflow"></span>
<img src="figs/SPICEY_wf.png" alt="SPICEY workflow. Differential single-cell RNA/ATAC data and region–gene annotations are used to compute specificity and entropy metrics, enabling the construction of tissue-specific regulatory networks." width="100%" height="100%" />
<p class="caption">
Figure 1: <em>SPICEY workflow</em>. Differential single-cell RNA/ATAC data and region–gene annotations are used to compute specificity and entropy metrics, enabling the construction of tissue-specific regulatory networks.
</p>
</div>

<div id="cell-type-specificity-indices" class="section level2">
<h2>Cell type specificity indices</h2>
<ul>
<li><strong>RETSI</strong> (Regulatory Element Tissue Specificity Index): quantifies cell-type specificity of regulatory regions.</li>
<li><strong>GETSI</strong> (Gene Expression Tissue Specificity Index): measures cell-type specificity of genes.</li>
</ul>
<p>Both metrics are based on weighted specificity scores that quantify how enriched each feature (gene or regulatory region) is within individual cell types, with higher values indicating greater tissue specificity:</p>
<p><span class="math display">\[
TSI_{i,x} = \frac{FC_{i,x}}{\max(FC)_x} \cdot w_{i,x}
\]</span></p>
<p><span class="math display">\[
w_{i,x} = \text{rescale}\left( -\log_{10}(padj_{i,x}),\ [0, 1] \right)
\]</span></p>
<p>where:</p>
<ul>
<li><span class="math inline">\(i\)</span> = cell type</li>
<li><span class="math inline">\(x \in \{r, g\}\)</span>, with <span class="math inline">\(r\)</span> = regulatory region; <span class="math inline">\(g\)</span> = gene</li>
<li><span class="math inline">\(\text{FC}_{i,x}\)</span> = fold-change of feature <span class="math inline">\(x\)</span> in cell type <span class="math inline">\(i\)</span></li>
</ul>
</div>
<div id="entropy-based-specificity-indices" class="section level2">
<h2>Entropy-based specificity Indices</h2>
<p>Measure the distributional skewness of feature activity/expression across cell types:</p>
<p><span class="math display">\[
H = - \sum_{i=1}^{N} p_i \log_2(p_i)
\]</span></p>
<p><span class="math display">\[
H_{norm} = 1 - e^{H}
\]</span> where:</p>
<ul>
<li><span class="math inline">\(i\)</span> = cell type</li>
<li><span class="math inline">\(p_{i,x} = \frac{a_{i,x}}{\sum_{j=1}^{N} a_{j,x}}\)</span> = activity/expression of feature <span class="math inline">\(x\)</span> across <span class="math inline">\(N\)</span> cell types</li>
<li><span class="math inline">\(H_{norm}\)</span> = normalized shannon entropy [0,1].</li>
</ul>
<p>Feature activity or expression is normalized across cell types to form a probability distribution, where low entropy indicates specificity to a few cell types, and high entropy reflects broad, uniform activity.</p>
<p><br></p>
</div>
<div id="region-to-gene-annotation" class="section level2">
<h2>Region-to-gene annotation</h2>
<p>To help construct tissue-specific gene-regulatory networks, SPICEY optionally annotates regulatory regions to their putative target genes using (1) <strong>nearest-gene</strong> (based on proximity to the closest TSS), or (2) <strong>co-accessibility</strong> (using correlated chromatin accessibility).</p>
</div>
</div>
<div id="results" class="section level1">
<h1>Results</h1>
<div id="spicey-reveals-cell-type-specific-regulatory-activity-in-human-pancreatic-islets" class="section level2">
<h2>SPICEY reveals cell-type specific regulatory activity in human pancreatic islets</h2>
<p>SPICEY scores from non-diabetic human pancreatic islets (HPAP) <span class="citation">Authors (2022)</span> showed distinct specificity patterns across cell types, highlighting the heterogeneous regulatory and transcriptional landscape of the pancreatic islets (<strong>Figure</strong> <a href="#fig:dist">2</a>).</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:dist"></span>
<img src="POSTER_EUROBIOC_20250805_files/figure-html/dist-1.png" alt="Distribution values reflect higher specificity in tissue-specific features (brown) compared to ubiquitous (green)" width="100%" height="100%" />
<p class="caption">
Figure 2: Distribution values reflect higher specificity in tissue-specific features (brown) compared to ubiquitous (green)
</p>
</div>

<p>Known tissue-specific promoters and enhancers have higher specificity than ubiquitous ones. Thus, SPICEY captures how cell-type-specific gene expression aligns with cell-specific cis-regulatory activity. (<strong>Figure</strong> <a href="#fig:markers">3</a>).</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:markers"></span>
<img src="POSTER_EUROBIOC_20250805_files/figure-html/markers-1.png" alt="Promoter and distal regions show higher SPICEY scores for tissue-specific genes." width="100%" height="100%" />
<p class="caption">
Figure 3: Promoter and distal regions show higher SPICEY scores for tissue-specific genes.
</p>
</div>

<p>Known ubiquitous sites have higher entropy -indicating widespread activity- while tissue-specific sites have significantly lower entropy -reflecting a bias towards cell-type-specific activity- (<strong>Figure</strong> <a href="#fig:entropy">4</a>), confirming that SPICEY effectively captures regulatory and transcriptional specificity in human pancreatic islets.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:entropy"></span>
<img src="POSTER_EUROBIOC_20250805_files/figure-html/entropy-1.png" alt="Entropy metrics reveal lower values for tissue-specific features, confirming stronger cell-type specificity." width="100%" height="100%" />
<p class="caption">
Figure 4: Entropy metrics reveal lower values for tissue-specific features, confirming stronger cell-type specificity.
</p>
</div>

</div>
<div id="spicey-uncovers-both-known-and-novel-markers-in-human-pancreatic-islets" class="section level2">
<h2>SPICEY uncovers both known and novel markers in human pancreatic islets</h2>
<p>SPICEY identified well-known markers in endocrine cells (<em>INS, IGF2, SST, IRX1/IRX2</em>) as well as less common ones like <em>ADCYAP1, SIX2</em>, and <em>LRFN5.</em></p>
<p>It recovered established markers such as <em>REG1A/B, PI3, KRT</em> family, and <em>COL6A1</em>, for non-endocrine cells while also capturing novel potential candidates like <em>GIMAP</em> genes in endothelial cells (<strong>Figure</strong> <a href="#fig:main2">5</a>).</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:main2"></span>
<img src="POSTER_EUROBIOC_20250805_files/figure-html/main2-1.png" alt="Top SPICEY cell-type-specific features. The heatmap displays the top 5 genes for each cell type based on RETSI and GETSI z-scores. Rows represent genes and columns represent cell types. Darker colors indicate higher specificity." width="100%" height="100%" />
<p class="caption">
Figure 5: <em>Top SPICEY cell-type-specific features.</em> The heatmap displays the top 5 genes for each cell type based on RETSI and GETSI z-scores. Rows represent genes and columns represent cell types. Darker colors indicate higher specificity.
</p>
</div>

</div>
</div>
<div id="conclusions" class="section level1">
<h1>Conclusions</h1>
<ul>
<li><p>SPICEY is an R package that quantifies cell-type specificity from single-cell multi-omic data using combined differential and entropy-based metrics, ideal for studying complex tissues using common single-cell workflows.</p></li>
<li><p>Identifies both, known and novel markers across well-defined and less-studied cell types, distinguishing tissue-specific from broadly active features.</p></li>
<li><p>Annotates regulatory elements to target genes helping elucidate tissue-specific regulatory networks.</p></li>
</ul>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<p><span class="citation">(<strong>R-posterdown?</strong>)</span>
<span class="citation">(<strong>R-rmarkdown?</strong>)</span></p>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-authors_p_h_pancdb_2022" class="csl-entry">
Authors, Authors, P. H. 2022. <span>“<span>PANCDB</span>.”</span> <em>PANCDB</em>. <a href="https://hpap.pmacs.upenn.edu/">https://hpap.pmacs.upenn.edu/</a>.
</div>
<div id="ref-lim_advances_2024" class="csl-entry">
Lim, Jongsu, Chanho Park, Minjae Kim, Hyukhee Kim, Junil Kim, and Dong-Sung Lee. 2024. <span>“Advances in Single-Cell Omics and Multiomics for High-Resolution Molecular Profiling.”</span> <em>Experimental &amp; Molecular Medicine</em> 56 (3): 515–26. <a href="https://doi.org/10.1038/s12276-024-01186-2">https://doi.org/10.1038/s12276-024-01186-2</a>.
</div>
</div>
</div>

</div>
<div class="main">
<p><img src="figs/main_spicey_white.png" style="width:85.0%" /></p>
</div>
<div class="main_bottom">
<img id="main-img-left" src=figs/UPF.png>
<img id="main-img-center" src=figs/STBG_LACAIXA.png>
<img id="main-img-right" src=figs/FUNDING_FINAL.png>
</div>
</div>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
var script = document.createElement("script");
script.type = "text/javascript";
var src = "true";
if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
if (location.protocol !== "file:" && /^https?:/.test(src))
src = src.replace(/^https?:/, '');
script.src = src;
document.getElementsByTagName("head")[0].appendChild(script);
})();
</script>


</body>
</html>
