---
poster_width: "33.1in"
poster_height: "46.8in"
main_topsize: 0.2 #percent coverage of the poster
main_bottomsize: 0.029
author:
  - name: '**Georgina Fuentes-Páez**'
    affil: 1
    main: true
    orcid: '0009-0003-9417-7148'
    email: georgina.fuentes@upf.edu
  - name: Nacho Molina
    affil: 2
  - name: Mireia Ramos-Rodríguez
    affil: 1
  - name: Lorenzo Pasquali
    affil: 1
affiliation:
  - num: 1
    address: Endocrine Regulatory Genomics, Department of Experimental & Health Sciences, University Pompeu Fabra, 08003, Barcelona, Spain.
  - num: 2
    address: Institut de génétique et de biologie moléculaire et cellulaire (IGBMC), University of Strasbourg, 67404, Illkirch, France
theme: default
#STYLE & FORMATTING
title_textsize: "66pt"
author_textsize: "42pt"
authorextra_textsize: "30pt"
affiliation_textsize: "21pt"
affiliation_textcol: '##2c2022'
caption_fontsize: "21pt"
#Middle of the poster
main_fontfamily: "Lato"
main_textcol: "##2c2022"
main_textaplha: 100
main_textsize: "80pt"
font_family: "Lato"
main_findings:
- "![](figs/TITLE_brown.svg){ width=200% }"
logoleft_name: figs/DEPT.CESt_blanc_heng.png
logoright_width: "0.05%" # Significantly reduced the size
logocenter_width: "0.05%" # Significantly reduced the size
# logocenter_name: figs/QR_SPICEY_GITHUB2.png
#---POSTER BODY OPTIONS---#
primary_colour: "#d4cbcc" #Main colour used for poster design.
accent_colour: "#ebe7e7" #Secondary colour use for poster design.
secondary_colour: "#f7f5f5" #A third colour option for adding some "pop" to the poster colour palette
body_bgcol: "#FFFFFF"
body_textsize: "28pt"
body_fontfamily: "Lato"
body_textcol: "##2c2022"
reference_textsize: "18pt"
output: 
  posterdown::posterdown_betterport:
    self_contained: false
    pandoc_args: --mathjax
    number_sections: false
logoright_name: figs/funding_ministerio.png
logoright_position: "top"
---

```{r setup, include=FALSE}

knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      tidy = FALSE,
                      message = FALSE,
                      fig.align = 'center',
                      out.width = "100%", 
                      results = "hold")
options(knitr.table.format = "html") 
Sys.setlocale(locale="C")

## Packages
library(GenomicRanges)
library(dplyr)
library(ggplot2)
library(Seurat)
library(tidyr)
library(regioneR)
library(purrr)
library(Signac)
library(purrr)
library(ggside)
library(viridis)
library(tibble)
library(ggsignif)
library(RColorBrewer)
library(ggplotify)
library(pheatmap)
library(patchwork)
library(scales)
library(data.table)
library(here)
library(S4Vectors)
library(ggpubr)
library(ggridges)
library(scales) 
library(tidyverse)
library(org.Hs.eg.db)
library(GenomicFeatures)


theme_set(theme_bw(base_size=15))


## spicey data 
SPICEY_ANNOTATED_COACC <- readRDS("/homes/users/gfuentes/scratch/projects/spicey_paper/HPAP_CTRL/data/HPAP_CTRL_SPICEY_COACC.rds")

## markers 
hkg <- fgsea::gmtPathways("/homes/users/gfuentes/scratch/projects/spicey_paper/data/HSIAO_HOUSEKEEPING_GENES.v2024.1.Hs.gmt")
ubiquitous <- hkg$HSIAO_HOUSEKEEPING_GENES

ct_markers <- readRDS("/homes/users/gfuentes/scratch/projects/spicey_paper/HPAP_CTRL/data/CT_MARKERS.rds")

cell_colors <- c("Alpha" = "#008E80",
                 "Ductal" = "#5A7A98",
                 "Beta" = "#D37972",
                 "Delta" = "#fcbbb6",
                 "Acinar" = "#B4609B",
                 "Gamma" = "#C6E6FF",
                 "Activated-stellate" = "#fbb679",
                 "Quiescent-stellate" = "#ffdd6c",
                 "Unknown" = "#c7c7c7",
                 "Immune" = "#7E9F66",
                 "Epsilon" = "#A3D2CA",
                 "Endothelial" = "#87abbf")

```

```{=html}
<style>
  #main-img-left, #main-img-center, #main-img-right {
    width: 10%;
  }
  #main-img-left {
    width: 20%;
  }
  /* Headers */
  h1{
    font-size: 42pt;
  }
  h2,h3,h4,h5,h6{
    font-size: 32pt;
  }
</style>
```

```{css, echo=FALSE}
.table caption {
    font-size: 24pt
}
```

# Overview

SPICEY (SPecificity Index for Coding and Epigenetic activitY) is an R package designed to quantify cell-type specificity from single cell multi-omic data.

## Cell type specificity indices

-   **RETSI** (Regulatory Element Tissue Specificity Index): quantifies cell-type specificity of regulatory regions.
-   **GETSI** (Gene Expression Tissue Specificity Index): measures cell-type specificity of genes.

Both metrics are based on weighted specificity scores that quantify how enriched each feature (gene or regulatory region) is within individual cell types: 

$$
TSI_{i,x} = \frac{FC_{i,x}}{\max(FC)_x} \cdot w_{i,x}
$$

$$
w_{i,x} = \text{rescale}\left( -\log_{10}(padj_{i,x}),\ [0, 1] \right)
$$

where:

-   $i$ = cell type
-   $x \in \{r, g\}$, with $r$ = regulatory region (RETSI); $g$ = gene (GETSI)
-   $\text{FC}_{i,x}$ = fold-change of feature $x$ in cell type $i$

## Entropy-based specificity Indices

To measure the distributional skewness of feature activity/expression across cell types:

$$
H = - \sum_{i=1}^{N} p_i \log_2(p_i)
$$

$$
H_{norm} = 1 - e^{H}
$$ where:

-   $i$ = cell type
-   $p_{i,x} = \frac{a_{i,x}}{\sum_{j=1}^{N} a_{j,x}}$ = activity or expression of feature $x$ across $N$ cell types
-   $H_{norm}$ = normalized shannon entropy [0,1].

Feature activity or expression is normalized across cell types to follow a probability distribution; low entropy means specific to few cell types, high entropy means spread out evenly.

## Region-to-gene annotation

Using (1) either **nearest-gene** (based on proximity to the closest TSS) or (2) **co-accessibility** (using correlated chromatin accessibility) to help construct tissue-specific gene-regulatory networks.

# Results

SPICEY scores from non-diabetic human pancreatic islets (HPAP) (**Figure** \@ref(fig:suplementary1)) revealed distinct specificity patterns across cell types, highlighting the heterogeneous regulatory and transcriptional landscape of the pancreatic islets (**Figure** \@ref(fig:main1)).

```{r suplementary1, fig.align='center', fig.cap="(ref:suplementary1)", fig.show='hold', message=FALSE, warning=FALSE, out.width="80%", out.height="80%"}
knitr::include_graphics("/homes/users/gfuentes/scratch/projects/spicey_paper/HPAP_CTRL/posters/figs/HPAP_UMAP_subset.png")
```

(ref:suplementary1) *Joint snATAC and scRNA UMAP of human pancreatic islet cell types* from 5 non-diabetic HPAP donors. Each point represents a nucleus and is colored by predicted cell type.

Tissue-specific promoters and enhancers have higher specificity than ubiquitous ones, showing that cell-type-specific gene expression aligns with cell-specific cis-regulatory activity.  (**Figure** \@ref(fig:main1)). Ubiquitous sites have higher entropy -indicating widespread activity- while tissue-specific sites have significantly lower entropy -reflecting a bias towards cell-type-specific activity- (**Figure** \@ref(fig:main1)), confirming that SPICEY effectively captures regulatory and transcriptional specificity in human pancreatic islets.

```{r main1, message=FALSE, warning=FALSE, fig.show="hold", echo=FALSE, fig.cap="(ref:main1)", fig.width=5, fig.height=9}

# 1. DISTRIBUTION ----
# Prepare long data for SPICEY metrics
spicey_long <- SPICEY_ANNOTATED_COACC$linked %>%
  as.data.frame() %>%
  group_by(gene_id) %>%
  # filter(!(all(RETSI == 0 | is.na(RETSI)) & all(GETSI == 0 | is.na(GETSI)))) %>%
  ungroup() %>%
  pivot_longer(cols = c(RETSI, GETSI), names_to = "metric", values_to = "value") %>%
  filter(!is.na(value))

# Compute combined mean (RETSI + GETSI) for ordering
cell_order <- spicey_long %>%
  group_by(cell_type) %>%
  summarise(mean_value = mean(value, na.rm = TRUE), .groups = "drop") %>%
  arrange(desc(mean_value)) %>%
  pull(cell_type)

# Reorder factor levels
spicey_long <- spicey_long %>%
  mutate(cell_type = factor(cell_type, levels = cell_order))

# Combine and summarize
.markers <- bind_rows(
  SPICEY_ANNOTATED_COACC$linked %>%
  data.frame() %>%
  inner_join(ct_markers %>% dplyr::select(-Source),
             by = c("gene_id" = "Gene", "cell_type" = "CellType")) %>%
  # filter(TSS_gene %in% ct_markers$Gene) %>%
  mutate(type = "Tissue-specific"), 

  SPICEY_ANNOTATED_COACC$linked %>%
  data.frame() %>%
  filter(gene_id %in% ubiquitous) %>%
  mutate(type = "Ubiquitous")
  ) %>%
  mutate(
    type = factor(type, levels = c("Ubiquitous", "Tissue-specific")),
    RETSI = as.numeric(RETSI),
    GETSI = as.numeric(GETSI)
  ) %>%
  filter(!is.na(RETSI), !is.na(GETSI)) %>%
  group_by(cell_type, type) %>%
  summarise(
    RETSI = mean(RETSI, na.rm = TRUE),
    GETSI = mean(GETSI, na.rm = TRUE),
    .groups = "drop"
  )

# Plot 1: Boxplot + jitter + marker dots
distribution <- ggplot(spicey_long, aes(x = cell_type, y = value)) +
  geom_jitter(color = "grey80", width = 0.15, size = 0.4, alpha = 0.05) +
  geom_boxplot(aes(fill = cell_type), position = position_dodge(width = 0.8),
               outlier.shape = NA, linewidth = 0.6, color = "gray45") +
  geom_point(
    data = .markers %>%
      pivot_longer(cols = c(RETSI, GETSI), names_to = "metric", values_to = "value"),
    aes(x = cell_type, y = value, color = type),
    size = 2
  ) +
  scale_color_manual(values = c(
  "Tissue-specific" = "#965819",
  "Ubiquitous" = "#585800"
)) +
  facet_grid(~metric) +
  scale_fill_manual(values = cell_colors) +
  labs(x = "Cell type", y = "SPICEY measure") +
  coord_flip() +
  theme_gray(base_size = 14) +
  theme(
    legend.position = "none",
    panel.grid.minor = element_blank(),
    axis.title = element_text(size = 10, margin = margin(t = 5)),
    axis.text = element_text(size = 8, margin = margin(t = 2)),
    axis.text.x = element_text(angle = 45, hjust = 1),
    strip.text = element_text(size = 8),
    plot.title = element_text(size = 12),
    plot.margin = margin(5, 2, 5, 5)
  )

distribution <- cowplot::plot_grid(distribution,                                  
                                   labels = c("A"), 
                                   label_size = 12)

# 2. MARKERS ----
prom <- bind_rows(
  # Tissue-specific
  SPICEY_ANNOTATED_COACC$linked %>%
    data.frame() %>%
    inner_join(ct_markers %>% dplyr::select(-Source),
               by = c("gene_id" = "Gene", "cell_type" = "CellType")) %>%
    mutate(type = "Tissue-specific"),
  
  # Ubiquitous
  SPICEY_ANNOTATED_COACC$linked %>%
    data.frame() %>%
    filter(gene_id %in% ubiquitous) %>%
    mutate(type = "Ubiquitous")
) %>%
  mutate(
    type = factor(type, levels = c("Ubiquitous", "Tissue-specific")),
    RETSI = as.numeric(RETSI),
    GETSI = as.numeric(GETSI)
  ) %>%
  filter(!is.na(RETSI), !is.na(GETSI), !is.na(type)) %>%
  pivot_longer(cols = c(RETSI, GETSI),
               names_to = "spicey_measure",
               values_to = "value") %>%
  # Apply conditional TSS filters *only for RETSI*
  filter(
    spicey_measure != "RETSI" |
      (type == "Tissue-specific" & TSS_gene %in% ct_markers$Gene & annotation == "Promoter" |
      (type == "Ubiquitous" & TSS_gene %in% ubiquitous & annotation == "Promoter")
  )) %>%
  mutate(
    color_key = paste(type, spicey_measure, sep = "_")
  ) %>%
  ggplot(aes(x = value, color = color_key, fill = color_key)) +
  geom_density(alpha = 0.6) +
  ggside::geom_xsideboxplot(aes(y = spicey_measure, fill = color_key), 
                            orientation = "y", outlier.size = 0.1, 
                            outlier.alpha = 0.5) +
  ggside::scale_xsidey_discrete() +
  scale_y_continuous(name = "Density") +
  scale_color_manual(values = c(
    "Ubiquitous_RETSI" = "#656d4a",
    "Ubiquitous_GETSI" = "#5e614b",
    "Tissue-specific_RETSI" = "#b37a56",
    "Tissue-specific_GETSI" = "#8c5e3c"
  )) +
  scale_fill_manual(values = c(
    "Ubiquitous_RETSI" = "#a4ac86",
    "Ubiquitous_GETSI" = "#919b74",
    "Tissue-specific_RETSI" = "#D2B48C",
    "Tissue-specific_GETSI" = "#c8a77b"
  )) +
  labs(x = "SPICEY measure") +
  theme_gray() +
  theme(
    legend.position = "none",
    ggside.panel.scale.x = 0.3,
    axis.title = element_text(size = 10, margin = margin(t = 5)),
    axis.text = element_text(size = 8, margin = margin(t = 2)),
    strip.text = element_text(size = 8),
    axis.text.x = element_text(angle = 30),
    plot.title = element_text(size = 12)
  ) +
  facet_grid(~ type) +
  ggtitle("Promoter")



enh <- bind_rows(
  # Tissue-specific: match by gene and cell type, keep all rows
  SPICEY_ANNOTATED_COACC$linked %>%
    data.frame() %>%
    # filter(direct_TSS == TRUE) %>%
    inner_join(ct_markers %>% dplyr::select(-Source),
               by = c("gene_id" = "Gene", "cell_type" = "CellType")) %>%
    mutate(type = "Tissue-specific") %>%
    filter(gene_id %in% ct_markers$Gene),
  
  # Ubiquitous: just filter by gene list
  SPICEY_ANNOTATED_COACC$linked %>%
    data.frame() %>%
    filter(gene_id %in% ubiquitous) %>%
    mutate(type = "Ubiquitous")) %>%
  
  # Factor level for consistency in plotting
  mutate(type = factor(type, levels = c("Ubiquitous", "Tissue-specific")),
         RETSI = as.numeric(RETSI),
         GETSI = as.numeric(GETSI)) %>%
  filter(!is.na(RETSI), !is.na(GETSI), !is.na(type)) %>%
  # mutate(region = paste0(seqnames, ":", start, "-", end)) %>%
  pivot_longer(cols = c(RETSI, GETSI),
               names_to = "spicey_measure",
               values_to = "value") %>% 
  filter(
    spicey_measure != "RETSI" |
    (annotation == "Distal")) %>%
  mutate(
    color_key = paste(type, spicey_measure, sep = "_")
  ) %>%
  ggplot(aes(x = value, color = color_key, fill = color_key)) +
  geom_density(alpha = 0.6) +
  ggside::geom_xsideboxplot(aes(y = spicey_measure, fill = color_key), 
                            orientation = "y", outlier.size = 0.1, 
                            outlier.alpha = 0.5) +
  ggside::scale_xsidey_discrete() +
  scale_y_continuous(name = "Density") +
  scale_color_manual(values = c(
    "Ubiquitous_RETSI" = "#656d4a",
    "Ubiquitous_GETSI" = "#5e614b",
    "Tissue-specific_RETSI" = "#b37a56",
    "Tissue-specific_GETSI" = "#8c5e3c"
  )) +
  scale_fill_manual(values = c(
    "Ubiquitous_RETSI" = "#a4ac86",
    "Ubiquitous_GETSI" = "#919b74",
    "Tissue-specific_RETSI" = "#D2B48C",
    "Tissue-specific_GETSI" = "#c8a77b"
  )) +
  
  labs(x = "SPICEY measure") +
  # ggtitle("Promoter regions") +
  theme_gray() +
  theme(
    legend.position = "none",
    ggside.panel.scale.x = 0.3,
    axis.title = element_text(size = 10, margin = margin(t = 5)),  # smaller top margin
    axis.text = element_text(size = 8, margin = margin(t = 2)),   # bring x-ticks closer
    strip.text = element_text(size = 8),
    axis.text.x = element_text(angle=30),
    plot.title = element_text(size = 12)
  ) +
  facet_grid(~ type) +
  ggtitle("Distal")


final <- cowplot::plot_grid(prom, enh,                                  
                            labels = c("B", "C"), 
                            label_size = 12)

# 3. ENTROPY ----
## getsi H
entropy_genes <- bind_rows(
  # Tissue-specific data
  SPICEY_ANNOTATED_COACC$linked %>%
    data.frame() %>%
    dplyr::filter(gene_id %in% ct_markers$Gene) %>%
    mutate(type = "Tissue-specific"),
  
  # Ubiquitous data
  SPICEY_ANNOTATED_COACC$linked %>%
    data.frame() %>%
    dplyr::filter(gene_id %in% ubiquitous) %>%
    mutate(type = "Ubiquitous")) %>%
  mutate(type = factor(type, levels = c("Ubiquitous", "Tissue-specific")),
         RETSI = as.numeric(RETSI),
         GETSI = as.numeric(GETSI)) %>%
  filter(
    !is.na(RETSI),
    !is.na(GETSI),
    !is.na(type)) %>%
  {rownames(.) <- NULL; .} %>%
  filter(!is.na(GETSI),
         !is.na(RETSI)) %>%
  pivot_longer(cols = c(GETSI_entropy),
               names_to = "spicey_measure",
               values_to = "value") %>%
  mutate(spicey_measure = gsub("_entropy", "", spicey_measure)) %>%
  ggplot(aes(x = type,
             y = value,
             fill = type)) +
  geom_violin() +
  geom_boxplot(alpha = 0.4) +  # Adjust outlier behavior if needed
  theme_gray() +
  labs(x = "", y = expression(H[norm]~"GETSI")) +
  theme(
    legend.position = "none",
    ggside.panel.scale.x = 0.3,
    axis.title = element_text(size = 10, margin = margin(t = 5)),
    axis.text = element_text(size = 8, margin = margin(t = 2)),
    axis.text.x = element_text(angle = 45, hjust = 1),
    strip.text = element_text(size = 8),
    plot.title = element_text(size = 12)
  ) +
  scale_fill_manual(values = c(
    "Ubiquitous" = "#919b74",
    "Tissue-specific" = "#c8a77b"
  )) +
  # ylim(c(1,1.2))+
  geom_signif(comparisons = list(
    c("Tissue-specific", "Ubiquitous")),
    y_position = c(1, 1.1),  # Adjust y position for annotations
    map_signif_level = TRUE,  # Display significance level
    textsize = 2.3,
    vjust = 0.01,               # Moves text slightly above the line
    tip_length = 0.02         # Negative value to flip arrows **upwards**
  )

g_entropy <- cowplot::plot_grid(entropy_genes,
                                labels = c("D"),
                                label_size = 12)

## retsi H
entropy_prom <- bind_rows(
  # Tissue-specific data
  SPICEY_ANNOTATED_COACC$linked %>%
    data.frame() %>%
    dplyr::filter(gene_id %in% ct_markers$Gene) %>%
    mutate(type = "Tissue-specific"),
  
  # Ubiquitous data
  SPICEY_ANNOTATED_COACC$linked %>%
    data.frame() %>%
    dplyr::filter(gene_id %in% ubiquitous) %>%
    mutate(type = "Ubiquitous")) %>%
  mutate(type = factor(type, levels = c("Ubiquitous", "Tissue-specific")),
         RETSI = as.numeric(RETSI),
         GETSI = as.numeric(GETSI)) %>%
  filter(
    !is.na(RETSI),
    !is.na(GETSI),
    !is.na(type)) %>%
  {rownames(.) <- NULL; .} %>%
  filter(!is.na(GETSI),
         !is.na(RETSI)) %>%
  pivot_longer(cols = c(RETSI_entropy),
               names_to = "spicey_measure",
               values_to = "value") %>%
  # Apply conditional TSS filters *only for RETSI*
  filter(
    spicey_measure != "RETSI" |
      (type == "Tissue-specific" & TSS_gene %in% ct_markers$Gene & annotation == "Promoter" |
         (type == "Ubiquitous" & TSS_gene %in% ubiquitous & annotation == "Promoter")
      )) %>%
  mutate(spicey_measure = gsub("_entropy", "", spicey_measure)) %>%
  ggplot(aes(x = type,
             y = value,
             fill = type)) +
  geom_violin() +
  geom_boxplot(alpha = 0.4) +  # Adjust outlier behavior if needed
  theme_gray() +
  labs(x = "", y = expression(H[norm]~"RETSI")) +
  theme(
    legend.position = "none",
    ggside.panel.scale.x = 0.3,
    axis.title = element_text(size = 10, margin = margin(t = 5)),
    axis.text = element_text(size = 8, margin = margin(t = 2)),
    axis.text.x = element_text(angle = 45, hjust = 1),
    strip.text = element_text(size = 8),
    plot.title = element_text(size = 12)
  ) +
  scale_fill_manual(values = c(
    "Ubiquitous" = "#919b74",
    "Tissue-specific" = "#c8a77b"
  )) +
  # ylim(c(1,1.2))+
  geom_signif(comparisons = list(
    c("Tissue-specific", "Ubiquitous")),
    y_position = c(1, 1.1),  # Adjust y position for annotations
    map_signif_level = TRUE,  # Display significance level
    textsize = 2.3,
    vjust = 0.01,               # Moves text slightly above the line
    tip_length = 0.02         # Negative value to flip arrows **upwards**
  ) +
  # facet_grid(~spicey_measure) +
  ggtitle("Promoter")


entropy_distal <- bind_rows(
  # Tissue-specific data
  SPICEY_ANNOTATED_COACC$linked %>%
    data.frame() %>%
    dplyr::filter(gene_id %in% ct_markers$Gene) %>%
    mutate(type = "Tissue-specific"),
  
  # Ubiquitous data
  SPICEY_ANNOTATED_COACC$linked %>%
    data.frame() %>%
    dplyr::filter(gene_id %in% ubiquitous) %>%
    mutate(type = "Ubiquitous")) %>%
  mutate(type = factor(type, levels = c("Ubiquitous", "Tissue-specific")),
         RETSI = as.numeric(RETSI),
         GETSI = as.numeric(GETSI)) %>%
  filter(
    !is.na(RETSI),
    !is.na(GETSI),
    !is.na(type)) %>%
  {rownames(.) <- NULL; .} %>%
  filter(!is.na(GETSI),
         !is.na(RETSI)) %>%
  pivot_longer(cols = c(RETSI_entropy),
               names_to = "spicey_measure",
               values_to = "value") %>%
  mutate(spicey_measure = gsub("_entropy", "", spicey_measure)) %>%
  # Apply conditional TSS filters *only for RETSI*
  filter(
    spicey_measure != "RETSI" |
      (annotation == "Distal")) %>%
  ggplot(aes(x = type,
             y = value,
             fill = type)) +
  geom_violin() +
  geom_boxplot(alpha = 0.4) +  # Adjust outlier behavior if needed
  theme_gray() +
  labs(x = "", y = "") +
  theme(
    legend.position = "none",
    ggside.panel.scale.x = 0.3,
    axis.title = element_text(size = 10, margin = margin(t = 5)),
    axis.text = element_text(size = 8, margin = margin(t = 2)),
    axis.text.x = element_text(angle = 45, hjust = 1),
    strip.text = element_text(size = 8),
    plot.title = element_text(size = 12)
  ) +
  scale_fill_manual(values = c(
    "Ubiquitous" = "#919b74",
    "Tissue-specific" = "#c8a77b"
  )) +
  # ylim(c(1,1.2))+
  geom_signif(comparisons = list(
    c("Tissue-specific", "Ubiquitous")),
    y_position = c(1, 1.1),  # Adjust y position for annotations
    map_signif_level = TRUE,  # Display significance level
    textsize = 2.3,
    vjust = 0.01,               # Moves text slightly above the line
    tip_length = 0.02         # Negative value to flip arrows **upwards**
  ) +
  # facet_grid(~spicey_measure) +
  ggtitle("Distal")


r_entropy <- cowplot::plot_grid(entropy_prom, entropy_distal,
                                labels = c("E"),
                                label_size = 12)

entropy_final <- cowplot::plot_grid(
  g_entropy, r_entropy,
  label_size = 12,
  rel_widths = c(0.57, 1),
  align = "h",
  axis = "tb"
)

# FINAL FIGURE ----
cowplot::plot_grid(
  distribution,
  final,
  entropy_final,
  nrow = 3,
  rel_heights = c(1, 1, 1.1))

```

(ref:main1) *SPICEY reveals cell-type-specificity.* (A) Distribution values reflect higher specificity in tissue-specific features (brown) compared to ubiquitous (green). (B–C) Promoter and distal regions show higher SPICEY scores for tissue-specific genes. (D–E) Entropy metrics reveal lower values for tissue-specific features, confirming stronger cell-type specificity.

SPICEY identified well-known markers in endocrine cells (*INS, IGF2, SST, IRX1/IRX2*) as well as less common ones like *ADCYAP1, SIX2*, and *LRFN5.* It recovered established markers such as *REG1A/B, PI3, KRT* family, and *COL6A1*, for non-endocrine cells while also highlighting novel candidates like *GIMAP* genes in endothelial cells (**Figure** \@ref(fig:main2)).

```{r main2, message=FALSE, warning=FALSE, fig.show="hold", echo=FALSE, fig.cap="(ref:main2)", fig.width=5, fig.height=9}

SPICEY::spicey_heatmap(df = SPICEY_ANNOTATED_COACC$linked, 
                       spicey_measure = c("SPICEY"), 
                       combined_zscore = FALSE, 
                       top_n = 5)

```

(ref:main2) *Top SPICEY cell-type-specific features.* The heatmap displays the top 5 genes for each cell type based on RETSI and GETSI z-scores. Rows represent genes and columns represent cell types. Darker colors indicate higher specificity. 

# Conclusions

* SPICEY is an R package that quantifies cell-type specificity from single-cell multi-omic data using combined differential and entropy-based metrics, making it ideal for studying complex tissues using common single-cell workflows.

* It identifies both known and novel markers across well-defined and less-studied cell types, distinguishing tissue-specific from broadly active features.

* Annotates regulatory elements to target genes helping uncover coordinated regulatory networks.
